src/ApplicationResolver.php

```diff
@@ -36,41 +36,36 @@ public static function resolve(): Application
     {
         $app = (new self)->createApplication();
 
-        $vendorDir = self::getVendorDir() ?? getcwd().DIRECTORY_SEPARATOR.'vendor';
-        $composerConfigPath = dirname($vendorDir).DIRECTORY_SEPARATOR.'composer.json';
-
-        if (file_exists($composerConfigPath)) {
-            self::$composer = json_decode((string) file_get_contents($composerConfigPath), true);
-            $namespace = (string) key(self::$composer['autoload']['psr-4']);
-            $serviceProviders = array_values(array_filter(self::getProjectClasses($namespace, $vendorDir), static function ($class) use (
-                $namespace
-            ) {
-                /** @var class-string $class */
-                return strpos($class, $namespace) === 0 && self::isServiceProvider($class);
-            }));
-
-            foreach ($serviceProviders as $serviceProvider) {
-                $app->register($serviceProvider);
-            }
+        $classLoaders = array_map('get_class', ClassLoader::getRegisteredLoaders());
+        $currentVendorDir = getcwd().DIRECTORY_SEPARATOR.'vendor';
+        if (is_dir($currentVendorDir) && !array_key_exists($currentVendorDir, $classLoaders)) {
+            $classLoaders[$currentVendorDir] = ClassLoader::class;
         }
 
-        return $app;
-    }
-
-    protected static function getVendorDir(): ?string
-    {
-        $reflector = new ReflectionClass(ClassLoader::class);
-        $classLoaderPath = $reflector->getFileName();
-        if ($classLoaderPath === false) {
-            return null;
-        }
+        foreach (array_keys($classLoaders) as $vendorDir) {
+            if (substr($vendorDir, 0, 5) === 'phar:') {
+                continue;
+            }
```

Comment:
What about just doing `$vendorDir = dirname($app->basePath(), 3)`?

Reply:
I'm just a janitor here.

`basePath` does not support custom loaders.

Adding a Laravel package to tests may help.

Reply:
For example there is a Composer plugin called `bin-dir`